lib/src/rules/directives_ordering.dart

```diff
@@ -229,6 +302,31 @@ class _Visitor extends SimpleAstVisitor {
         .forEach(reportDirective);
   }
 
+  void _checkSectionInOrder(
+      Set<AstNode> lintedNodes, Iterable<NamespaceDirective> nodes) {
+    void reportDirective(NamespaceDirective directive) {
+      if (lintedNodes.add(directive)) {
+        rule._reportLintWithDirectiveSectionOrderedAlphabeticallyMessage(
+            directive);
+      }
+    }
+
+    List<NamespaceDirective> addIfUpperBound(
+        List<NamespaceDirective> acc, NamespaceDirective curr) {
+      if (acc.isEmpty || acc.last.uriContent.compareTo(curr.uriContent) < 0) {
+        acc.add(curr);
+      }
+
+      return acc;
+    }
+
+    final inOrderNodes = nodes.fold(<NamespaceDirective>[], addIfUpperBound);
+
+    nodes.toSet()
+      ..removeAll(inOrderNodes)
+      ..forEach(reportDirective);
```

Comment:
You are all right, it would be better, but we couldn't figure it out of how to get the notInOrderNodes with a simple and readable fold.

To know if a directive node is in order, it is necessary to compare it with all nodes that are before it, if its uri content is higher than all of them then it is in order, so we needed to know which node is the maximum node before a node to know if it is in order or not, to do that it was easier to keep all the nodes in order with a ford and compare it just with the last one.

PTAL to the method addIfUpperBound